From 95d6d0f598dfb0b17b3c148a7bbbc4010afc34bb Mon Sep 17 00:00:00 2001
From: "Miss Islington (bot)"
 <31488909+miss-islington@users.noreply.github.com>
Date: Wed, 26 Aug 2026 02:59:49 +0200
Subject: [PATCH 5/5] [3.13] gh-156369: Update OpenSSL error ID data
 (GH-156380) (#156392)

gh-156369: Update OpenSSL error ID data (GH-156380)
(cherry picked from commit 7cc333439b6426a43cdbca690d9170c9db0d8669)

Co-authored-by: Zachary Ware <zach@python.org>
(cherry picked from commit 1c826a40a2f9606e284fd3d6d5f7c7c6732327ed)
---
 Modules/_ssl_data_300.h | 59 +++++++++++++++++++++++++++++++++++++----
 Modules/_ssl_data_40.h  |  9 +++++--
 2 files changed, 61 insertions(+), 7 deletions(-)

diff --git a/Modules/_ssl_data_300.h b/Modules/_ssl_data_300.h
index dc66731f6b6..39b637a8f6a 100644
--- a/Modules/_ssl_data_300.h
+++ b/Modules/_ssl_data_300.h
@@ -1,4 +1,9 @@
-/* File generated by Tools/ssl/make_ssl_data.py *//* Generated on 2023-06-01T03:03:52.163218 */
+/* File generated by Tools/ssl/make_ssl_data.py */
+/* Generated on 2026-08-25T19:28:21.667000+00:00 */
+/* Generated from Git commit openssl-3.0.22-0-ga279090b9 */
+/* Manually edited to retain definitions from 1.1.1 (GH-105174) */
+
+/* generated from args.lib2errnum */
 static struct py_ssl_library_code library_codes[] = {
 #ifdef ERR_LIB_ASN1
     {"ASN1", ERR_LIB_ASN1},
@@ -165,10 +170,10 @@ static struct py_ssl_library_code library_codes[] = {
 #ifdef ERR_LIB_X509V3
     {"X509V3", ERR_LIB_X509V3},
 #endif
-    { NULL }
+    {NULL, 0}  /* sentinel */
 };
 
-
+/* generated from args.reasons */
 static struct py_ssl_error_code error_codes[] = {
   #ifdef ASN1_R_ADDING_OBJECT
     {"ADDING_OBJECT", ERR_LIB_ASN1, ASN1_R_ADDING_OBJECT},
@@ -1535,6 +1540,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"CERTIFICATE_VERIFY_ERROR", 46, 100},
   #endif
+  #ifdef CMS_R_CIPHER_AEAD_IN_ENVELOPED_DATA
+    {"CIPHER_AEAD_IN_ENVELOPED_DATA", ERR_LIB_CMS, CMS_R_CIPHER_AEAD_IN_ENVELOPED_DATA},
+  #else
+    {"CIPHER_AEAD_IN_ENVELOPED_DATA", 46, 182},
+  #endif
   #ifdef CMS_R_CIPHER_AEAD_SET_TAG_ERROR
     {"CIPHER_AEAD_SET_TAG_ERROR", ERR_LIB_CMS, CMS_R_CIPHER_AEAD_SET_TAG_ERROR},
   #else
@@ -1640,6 +1650,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"ERROR_SETTING_RECIPIENTINFO", 46, 116},
   #endif
+  #ifdef CMS_R_ERROR_UNSUPPORTED_STATIC_KEY_AGREEMENT
+    {"ERROR_UNSUPPORTED_STATIC_KEY_AGREEMENT", ERR_LIB_CMS, CMS_R_ERROR_UNSUPPORTED_STATIC_KEY_AGREEMENT},
+  #else
+    {"ERROR_UNSUPPORTED_STATIC_KEY_AGREEMENT", 46, 196},
+  #endif
   #ifdef CMS_R_ESS_SIGNING_CERTID_MISMATCH_ERROR
     {"ESS_SIGNING_CERTID_MISMATCH_ERROR", ERR_LIB_CMS, CMS_R_ESS_SIGNING_CERTID_MISMATCH_ERROR},
   #else
@@ -1960,6 +1975,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"UNSUPPORTED_RECIPIENT_TYPE", 46, 154},
   #endif
+  #ifdef CMS_R_UNSUPPORTED_SIGNATURE_ALGORITHM
+    {"UNSUPPORTED_SIGNATURE_ALGORITHM", ERR_LIB_CMS, CMS_R_UNSUPPORTED_SIGNATURE_ALGORITHM},
+  #else
+    {"UNSUPPORTED_SIGNATURE_ALGORITHM", 46, 195},
+  #endif
   #ifdef CMS_R_UNSUPPORTED_TYPE
     {"UNSUPPORTED_TYPE", ERR_LIB_CMS, CMS_R_UNSUPPORTED_TYPE},
   #else
@@ -2085,6 +2105,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"RECURSIVE_DIRECTORY_INCLUDE", 14, 111},
   #endif
+  #ifdef CONF_R_RECURSIVE_SECTION_REFERENCE
+    {"RECURSIVE_SECTION_REFERENCE", ERR_LIB_CONF, CONF_R_RECURSIVE_SECTION_REFERENCE},
+  #else
+    {"RECURSIVE_SECTION_REFERENCE", 14, 126},
+  #endif
   #ifdef CONF_R_RELATIVE_PATH
     {"RELATIVE_PATH", ERR_LIB_CONF, CONF_R_RELATIVE_PATH},
   #else
@@ -2570,6 +2595,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"PEER_KEY_ERROR", 5, 111},
   #endif
+  #ifdef DH_R_Q_TOO_LARGE
+    {"Q_TOO_LARGE", ERR_LIB_DH, DH_R_Q_TOO_LARGE},
+  #else
+    {"Q_TOO_LARGE", 5, 130},
+  #endif
   #ifdef DH_R_SHARED_INFO_ERROR
     {"SHARED_INFO_ERROR", ERR_LIB_DH, DH_R_SHARED_INFO_ERROR},
   #else
@@ -7480,6 +7510,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"TLSV1_ALERT_INTERNAL_ERROR", 20, 1080},
   #endif
+  #ifdef SSL_R_TLSV1_ALERT_NO_APPLICATION_PROTOCOL
+    {"TLSV1_ALERT_NO_APPLICATION_PROTOCOL", ERR_LIB_SSL, SSL_R_TLSV1_ALERT_NO_APPLICATION_PROTOCOL},
+  #else
+    {"TLSV1_ALERT_NO_APPLICATION_PROTOCOL", 20, 1120},
+  #endif
   #ifdef SSL_R_TLSV1_ALERT_NO_RENEGOTIATION
     {"TLSV1_ALERT_NO_RENEGOTIATION", ERR_LIB_SSL, SSL_R_TLSV1_ALERT_NO_RENEGOTIATION},
   #else
@@ -7500,6 +7535,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"TLSV1_ALERT_UNKNOWN_CA", 20, 1048},
   #endif
+  #ifdef SSL_R_TLSV1_ALERT_UNKNOWN_PSK_IDENTITY
+    {"TLSV1_ALERT_UNKNOWN_PSK_IDENTITY", ERR_LIB_SSL, SSL_R_TLSV1_ALERT_UNKNOWN_PSK_IDENTITY},
+  #else
+    {"TLSV1_ALERT_UNKNOWN_PSK_IDENTITY", 20, 1115},
+  #endif
   #ifdef SSL_R_TLSV1_ALERT_USER_CANCELLED
     {"TLSV1_ALERT_USER_CANCELLED", ERR_LIB_SSL, SSL_R_TLSV1_ALERT_USER_CANCELLED},
   #else
@@ -8405,11 +8445,21 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"CRL_ALREADY_DELTA", 11, 127},
   #endif
+  #ifdef X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", ERR_LIB_X509, X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH},
+  #else
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", 11, 147},
+  #endif
   #ifdef X509_R_CRL_VERIFY_FAILURE
     {"CRL_VERIFY_FAILURE", ERR_LIB_X509, X509_R_CRL_VERIFY_FAILURE},
   #else
     {"CRL_VERIFY_FAILURE", 11, 131},
   #endif
+  #ifdef X509_R_DUPLICATE_ATTRIBUTE
+    {"DUPLICATE_ATTRIBUTE", ERR_LIB_X509, X509_R_DUPLICATE_ATTRIBUTE},
+  #else
+    {"DUPLICATE_ATTRIBUTE", 11, 140},
+  #endif
   #ifdef X509_R_ERROR_GETTING_MD_BY_NID
     {"ERROR_GETTING_MD_BY_NID", ERR_LIB_X509, X509_R_ERROR_GETTING_MD_BY_NID},
   #else
@@ -8580,6 +8630,5 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"WRONG_TYPE", 11, 122},
   #endif
-    { NULL }
+    {NULL, 0, 0}  /* sentinel */
 };
-
diff --git a/Modules/_ssl_data_40.h b/Modules/_ssl_data_40.h
index ea615f655bb..6d8631df0a2 100644
--- a/Modules/_ssl_data_40.h
+++ b/Modules/_ssl_data_40.h
@@ -1,6 +1,6 @@
 /* File generated by Tools/ssl/make_ssl_data.py */
-/* Generated on 2026-04-15T09:18:34.696139+00:00 */
-/* Generated from Git commit v3.15.0a8-132-gd14e31ed683 */
+/* Generated on 2026-08-25T19:25:20.590415+00:00 */
+/* Generated from Git commit openssl-4.0.2-0-gf089acdf4 */
 
 /* generated from args.lib2errnum */
 static struct py_ssl_library_code library_codes[] = {
@@ -9163,6 +9163,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"CRL_ALREADY_DELTA", 11, 127},
   #endif
+  #ifdef X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", ERR_LIB_X509, X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH},
+  #else
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", 11, 147},
+  #endif
   #ifdef X509_R_CRL_VERIFY_FAILURE
     {"CRL_VERIFY_FAILURE", ERR_LIB_X509, X509_R_CRL_VERIFY_FAILURE},
   #else
-- 
2.53.0

